File: app/src/main/java/com/eftimoff/bakingapp/app/view/BaseFragment.java
package com.eftimoff.bakingapp.app.view;

import android.content.Context;
import android.os.Bundle;
import android.support.annotation.LayoutRes;
import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.eftimoff.bakingapp.app.BakingApplication;
import com.eftimoff.bakingapp.app.injection.AppComponent;

import butterknife.ButterKnife;

public abstract class BaseFragment extends Fragment {

    @Nullable
    @Override
    public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        View view = inflater.inflate(getLayoutRes(), container, false);
        ButterKnife.bind(this, view);
        return view;
    }

    @Override
    public void onAttach(Context context) {
        super.onAttach(context);
        inject(BakingApplication.component());
    }

    @LayoutRes
    protected abstract int getLayoutRes();

    protected abstract void inject(AppComponent appComponent);

}

File: app/src/main/java/com/eftimoff/bakingapp/app/view/BaseFragment.kt
package com.eftimoff.bakingapp.app.view

import android.arch.lifecycle.LifecycleFragment
import android.content.Context
import android.os.Bundle
import android.support.annotation.LayoutRes
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import com.eftimoff.bakingapp.app.BakingApplication
import com.eftimoff.bakingapp.app.injection.AppComponent

abstract class BaseFragment : LifecycleFragment() {

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        val view = inflater.inflate(getLayoutRes(), container, false)
        return view
    }

    override fun onAttach(context: Context?) {
        super.onAttach(context)
        inject(BakingApplication.component)
    }

    @LayoutRes
    abstract fun getLayoutRes(): Int

    abstract fun inject(appComponent: AppComponent)

}